pkg/chain/ethereum/connect.go

```diff
@@ -368,3 +369,29 @@ func fetchChainConfig(ec *ethereumChain) (*relaychain.Config, error) {
 		RelayEntryTimeout:          relayEntryTimeout.Uint64(),
 	}, nil
 }
+
+func BalanceMonitor(chain chain.Handle) (*ethutil.BalanceMonitor, error) {
```

Comment:
I do not think this is the correct approach. We need to cast the interface to `*ethereumChain` so there is an assumption about the underlying type. Also, since we are creating Ethereum balance monitor, only the specific Ethereum implementation should be allowed as a parameter. I'd rather return the balance monitor pointer as a second parameter from `ethereum.Connect`.

Reply:
But we're in the `ethereum` package. Also, it's not casting but type assertion. We receive a generic chain, we check wheter it's ethereum and if yes, return the balance monitor. I would like o to avoid returning I from `connect` as balance monitoring is optional.

Reply:
Also, looks like it's a temporary inconvenience. Please see https://github.com/keep-network/keep-core/pull/2344#discussion_r582930532

Reply:
I am not sure this is temporary, it depends on where the connector will live. It still feels squirmy to me and IMO is an indicator that something is wrong with our abstractions. I think we should either return a concrete type from `Connect()` function or if we want to return `chain.Handle` return chain-specific extensions by either expanding `chain.Handle` or returning them as second value from `Connect` function.

Balance monitor is not triggered until `Observe` function is called. Another option could be to make connector a chain-specific concept and pass the configuration to `Connect` function. This latter can be more complex given that there is no single chain-specific session for ECDSA client.

Reply:
After an out-of-band discussion with @Shadowfiend, the last suggested option seems to us the best one, and given that https://github.com/keep-network/keep-ecdsa/pull/693 proposes to move tBTC system address from tBTC extension section into Ethereum configuration section, it should be enough to pass Ethereum configuration to `ethereum.Connect` function.

It plays well with build flags. Instead of having too much initialization logging in build-flag-dependent `cmd` package connector code, we just push it all behind chain's `Connect` facade that will do the right thing based on the configuration provided.

Reply:
Please see https://github.com/keep-network/keep-core/commit/5ac723255980b02ad039e199108b0de8202e2ece